app/api/routes/user.py

```diff
@@ -10,29 +19,241 @@
+        )
+
+
+@router.put("/{user_id}", response_model=UpdatedUserResponseModel)
+async def update_user(
+    user_id: UUID,
+    user_update: UpdateUserRequestModel,
+    current_user: User = Depends(get_current_user),
+):
+    # Step 1 & 2: Authenticate User and Validate User ID
+    if str(current_user.get("id")) != str(user_id):
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Access denied. You can only update your own details.",
+        )
+
+    # Retrieve the current user data
+    user_data = users_db.get(str(user_id))
+
+    if not user_data:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="User not found"
+        )
+
+    # Step 3: Update User Data
+    update_data = user_update.dict()
+
+    # Check if email is being updated and if it's unique
+    if any(u.get("email") == user_update.email for u in users_db.values()):
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT, detail="Email already registered"
+        )
+
+    if "password" in update_data:
+        # Step 4: Hash Password
+        user_data["hashed_password"] = get_password_hash(update_data.pop("password"))
+
+    # Update other fields
+    for field, value in update_data.items():
+        if value is not None:
+            user_data[field] = value
+
+    # Update the 'updated_at' timestamp
+    user_data["updated_at"] = datetime.now(timezone.utc)
+
+    # Update the user in the database
+    users_db[str(user_id)] = user_data
+
+    # Step 5: Return Updated User Data
+    updated_user = User(**user_data)
+    return UpdatedUserResponseModel(
+        id=updated_user.id,
+        username=updated_user.username,
+        email=updated_user.email,
+        is_admin=updated_user.is_admin,
+        is_active=updated_user.is_active,
+        created_at=updated_user.created_at,
+        updated_at=updated_user.updated_at,
+    )
+
+
+@router.get("/{user_id}", response_model=CreateUserResponseModel)
+async def get_user_details(
+    user_id: UUID, current_user: User = Depends(get_current_user)
+):
+    # Step 1: Authenticate User
+    if not current_user.get("is_admin") and str(current_user.get("id")) != str(user_id):
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Access denied. You can only view your own details.",
+        )
+
+    # Step 2: Validate User ID
+    user_data = users_db.get(str(user_id))
+    if not user_data:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="User not found"
+        )
+
+    # Step 3: Retrieve User Data
+    user = User(**user_data)
+
+    # Step 4: Return User Data
+    return CreateUserResponseModel(
+        id=user.id,
+        username=user.username,
+        email=user.email,
+        is_admin=user.is_admin,
+        is_active=user.is_active,
+        created_at=user.created_at,
+        updated_at=user.updated_at,
```

Comment:
The CreateUserResponseModel does not include the updated_at field, so that value will not be returned.

Reply:
done